core/src/test/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContextTest.java

```diff
@@ -0,0 +1,119 @@
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.metadata.NodeStateListener;
+import com.datastax.oss.driver.api.core.metadata.schema.SchemaChangeListener;
+import com.datastax.oss.driver.api.core.tracker.RequestTracker;
+import com.datastax.oss.driver.api.core.type.codec.TypeCodec;
+import com.datastax.oss.driver.internal.core.DefaultMavenCoordinates;
+import com.datastax.oss.driver.shaded.guava.common.collect.Lists;
+import com.datastax.oss.driver.shaded.guava.common.collect.Maps;
+import com.datastax.oss.protocol.internal.request.Startup;
+import com.datastax.oss.protocol.internal.util.collection.NullAllowingImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public class DefaultDriverContextTest {
+
+  private final MavenCoordinates driverProperties =
+      DefaultMavenCoordinates.buildFromResource(
+          DefaultDriverContextTest.class.getResource("/com/datastax/oss/driver/Driver.properties"));
+
+  private DefaultDriverContext defaultDriverContext;
+
+  // Mocks for instantiating the default driver context
+  @Mock private DriverConfigLoader configLoader;
+  List<TypeCodec<?>> typeCodecs = Lists.newArrayList();
+  @Mock private NodeStateListener nodeStateListener;
+  @Mock private SchemaChangeListener schemaChangeListener;
+  @Mock private RequestTracker requestTracker;
+  Map<String, Predicate<Node>> nodeFilters = Maps.newHashMap();
+  @Mock private ClassLoader classLoader;
+  @Mock private DriverConfig driverConfig;
+  @Mock private DriverExecutionProfile defaultProfile;
+
+  @Before
+  public void before() {
+    MockitoAnnotations.initMocks(this);
+    Mockito.when(configLoader.getInitialConfig()).thenReturn(driverConfig);
+    Mockito.when(driverConfig.getDefaultProfile()).thenReturn(defaultProfile);
+    defaultDriverContext =
+        new DefaultDriverContext(
+            configLoader,
+            typeCodecs,
+            nodeStateListener,
+            schemaChangeListener,
+            requestTracker,
+            nodeFilters,
+            classLoader,
+            null);
+  }
+
+  private void assertDefaultStartupOptions(Startup startup) {
+    assertThat(startup.getOptions()).containsEntry(Startup.CQL_VERSION_KEY, "3.0.0");
+    assertThat(startup.getOptions())
+        .containsEntry(InternalStartupOptions.DRIVER_NAME_KEY, driverProperties.getName());
+    assertThat(startup.getOptions()).containsKey(InternalStartupOptions.DRIVER_VERSION_KEY);
+    Version version =
+        Version.parse(startup.getOptions().get(InternalStartupOptions.DRIVER_VERSION_KEY));
+    // ensure it's a 4.x version, even if a pre-release
+    assertThat(version).isEqualByComparingTo(driverProperties.getVersion());
+  }
+
+  @Test
+  public void test_default_startup_options() {
```

Comment:
Nit: could you use the `should_` convention for consistency with the rest of the codebase? For example:
`should_build_minimal_startup_options`
`should_build_startup_options_with_compression`
`should_build_startup_options_with_custom_options`

And I would also cover the cases where the custom options contain one of the default ones or the compression.

Reply:
Yes, I forgot about this from my last PR. I need to break my old habit of test method names.